amp-pwa-reader/src/js/Article.js

```diff
@@ -174,36 +219,51 @@ class Article {
   }
 
   render() {
+      // Create an empty container for the AMP page
+      this.container = this.createShadowRoot();
+
+      // Tell Shadow AMP to initialize the AMP page in prerender-mode
+      this.ampDoc = AMP.attachShadowDoc(this.container, this.doc, this.url);
+      this.ampDoc.setVisibilityState('prerender');
 
-    // Create an empty container for the AMP page
+      return this.ampDoc.ampdoc.whenReady();
+  }
+
+  // in the streaming case, simply create the shadow root, and do the rest in stream()
+  renderStreaming() {
     this.container = this.createShadowRoot();
+    this.ampDoc = AMP.attachShadowDocAsStream(this.container, this.url); 
+  }
 
-    // Tell Shadow AMP to initialize the AMP page in prerender-mode
-    this.ampDoc = AMP.attachShadowDoc(this.container, this.doc, this.url);
-    this.ampDoc.setVisibilityState('prerender');
 
-    return this.ampDoc.ampdoc.whenReady();
+// We need to clone the featured image into the Shadow DOM so it scrolls
+// along. There are cases were we don't have a linked card from the list
+// view (e.g. we load directly into the article), in which case we need to
+// generate a new one.
+  getClonedCardElem() {
+    this.card ? this.cloneCard() : this.generateCard();
 
+    if (!this.streaming) {
+      this.clonedCardElem.lastElementChild.onclick = function () { return false; };  //TODO: we will lose this in streaming case  
```

Comment:
maybe explain why this is needed in the first place?

Reply:
Since the `lastElementChild` is AFAIK a link to a Guardian article, I'd figured the idea is to prevent users from accidentally finding themselves navigating to theguardian.com.

On the other hand, in the streaming case, I couldn't easily do this because the cloned card wasn't in the DOM yet at this point.  I finally left it as a TODO, because this didn't seem to affect the functionality.

I actually don't know why...